warehouse/snowflake/snowflake.go

```diff
@@ -355,17 +355,31 @@ func (sf *HandleT) loadTable(tableName string, columnMap map[string]string) (err
 		additionalJoinClause = fmt.Sprintf(`AND original.%[1]s = staging.%[1]s`, "TABLE_NAME")
 	}
 
+	stagingSelectColumns := "*"
+	if tableName == "USERS" {
+		userColMap := sf.CurrentSchema["USERS"]
+		var userColNames, firstValProps []string
+		for colName := range userColMap {
+			if colName == "ID" {
+				continue
+			}
+			userColNames = append(userColNames, colName)
+			firstValProps = append(firstValProps, fmt.Sprintf(`FIRST_VALUE(%[1]s IGNORE NULLS) OVER (PARTITION BY id ORDER BY uuid_ts DESC ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING) AS %[1]s`, colName))
+		}
+		stagingSelectColumns = fmt.Sprintf(`ID, %v`, strings.Join(firstValProps, ","))
+	}
+
 	sqlStatement = fmt.Sprintf(`MERGE INTO %[1]s AS original
 									USING (
 										SELECT * FROM (
-											SELECT *, row_number() OVER (PARTITION BY %[8]s ORDER BY RECEIVED_AT ASC) AS _rudder_staging_row_number FROM %[2]s
+											SELECT %[9]s, row_number() OVER (PARTITION BY %[8]s ORDER BY RECEIVED_AT DESC) AS _rudder_staging_row_number FROM %[2]s
```

Comment:
This will only merge values in staging, ignoring values that might already be in USERS table. If a user id has NULL values for a property in all staging rows, it will overwrite any values in USERS table. In redshift, this was solved using UNION